app/server/models/exchange.py

```diff
@@ -40,7 +41,27 @@ class ExchangeContract(ModelBase):
         lazy=True, foreign_keys='TransferAccount.exchange_contract_id'
     )
 
-    def add_token(self, token):
+    def get_subexchange_details(self, token_address):
+        if self.subexchange_address_mapping is None:
+            return None
+        return self.subexchange_address_mapping.get(token_address, None)
+
+    def _add_subexchange(self, token_address, subexchange_address, subexchange_reserve_ratio_ppm):
```

Comment:
given the 'subexchange' is technically another smart contract, we could store these as objects in their own rights, but the relationships would become pretty complicated

Reply:
yeah exactly -  token exchanges are actually made up of a network of linked smart contracts, and u need to know some details of those contracts

Reply:
yeahh i didn't understand subexchange either originally. is there a better terminology for it? or maybe just add a comment explaining it

Reply:
added a comment at the top of the class to explain :)